src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -468,6 +468,10 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
                 {
                     await Authenticate(username, timeout, async, cancellationToken);
 
+#if DEBUG
```

Comment:
Rather than wrapping each call with `#if DEBUG`, the methods themselves can have `[Conditional("DEBUG")]`. This automatically does what we want, and makes the call sites much nicer.